apps/assets/api/asset/asset.py

```diff
@@ -231,9 +235,12 @@ class AssetsTaskMixin:
     def perform_assets_task(self, serializer):
         data = serializer.validated_data
         assets = data.get("assets", [])
+        port = data.get("port", 22)
 
         if data["action"] == "refresh":
             task = update_assets_hardware_info_manual(assets)
+        elif data["action"] == "test_port":
+            task = test_assets_port_connectivity_manual(assets, port)
         else:
             asset = assets[0]
             if not asset.auto_config['ansible_enabled'] or \
```

Comment:
The code you provided contains no changes that require manual review. It appears to be functioning perfectly without any apparent errors or inefficiencies.

Potential optimizations could include:

- Use the `data` object directly instead of assigning it in each function call.
- Instead of initializing an empty list (`"assets": []`) at line 231 and then adding elements, use `.append()` method which is more concise:
   ```python
   assets = []
   assets.append(data.get("asset"))
``` 

But these aren't major improvements over existing lines and do not introduce significant complexity into the existing structure of the code snippet.